programs/stake/src/stake_state.rs

```diff
@@ -757,33 +761,71 @@ impl<'a> StakeAccount for KeyedAccount<'a> {
             match self.state()? {
                 StakeState::Stake(meta, mut stake) => {
                     meta.authorized.check(signers, StakeAuthorize::Staker)?;
+                    let split_rent_exempt_reserve = calculate_split_rent_exempt_reserve(
+                        meta.rent_exempt_reserve,
+                        self.data_len()? as u64,
+                        split.data_len()? as u64,
+                    );
 
                     // verify enough lamports for rent in new stake with the split
                     if split.lamports()? + lamports < meta.rent_exempt_reserve
```

Comment:
instead of `split_rent_exempt_reserve > meta.rent_exempt_reserve` below, the condition can be tweaked to `split.lamports()? + lamports < split_rent_exempt_reserve`?